services/browser-service/src/browser_service/browser_manager.py

```
@@ -66,41 +225,71 @@ async def navigate(
-
-            context = await self.browser.new_context(**context_options)
-
-            # Set cookies if provided
-            if cookies:
-                await context.add_cookies(
-                    [{"name": k, "value": v, "url": url} for k, v in cookies.items()]
-                )
+            # Use existing session or create temporary context
+            if session_id:
+                if session_id not in self._sessions:
+                    raise ValueError(f"Session not found: {session_id}")
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)